File: src/test/java/info/szadkowski/contact/properties/ThrottlingPropertiesTest.java
package info.szadkowski.contact.properties;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.TestPropertySource;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import java.time.Duration;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(SpringExtension.class)
@ContextConfiguration(classes = ThrottlingProperties.class)
@TestPropertySource(properties = {
        "throttling.clearExpiredRate=5s",
        "throttling.ip.limit=1",
        "throttling.ip.window=5s",
        "throttling.all.limit=2",
        "throttling.all.window=10s",
})
@EnableConfigurationProperties
class ThrottlingPropertiesTest {

  @Test
  void shouldInjectRecipientMail(@Autowired ThrottlingProperties properties) {
    assertThat(properties).isNotNull();
    assertThat(properties.getClearExpiredRate()).isEqualTo(Duration.ofSeconds(5));
    assertThat(properties.getIp().getLimit()).isEqualTo(1);
    assertThat(properties.getIp().getWindow()).isEqualTo(Duration.ofSeconds(5));
    assertThat(properties.getAll().getLimit()).isEqualTo(2);
    assertThat(properties.getAll().getWindow()).isEqualTo(Duration.ofSeconds(10));
  }
}

File: src/test/kotlin/info/szadkowski/contact/properties/ThrottlingPropertiesTest.kt
package info.szadkowski.contact.properties

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.TestPropertySource
import org.springframework.test.context.junit.jupiter.SpringExtension
import strikt.api.expectThat
import strikt.assertions.isEqualTo
import strikt.assertions.isNotNull
import java.time.Duration

@ExtendWith(SpringExtension::class)
@ContextConfiguration
@TestPropertySource(
    properties = [
        "throttling.clearExpiredRate=5s",
        "throttling.ip.limit=1",
        "throttling.ip.window=5s",
        "throttling.all.limit=2",
        "throttling.all.window=10s"
    ]
)
@EnableConfigurationProperties(ThrottlingProperties::class)
class ThrottlingPropertiesTest {

    @Test
    fun `Should inject throttling configuration`(@Autowired p: ThrottlingProperties) {
        expectThat(p) {
            get { clearExpiredRate }.isEqualTo(Duration.ofSeconds(5))
            get { ip }
                .isNotNull()
                .and {
                    get { limit }.isEqualTo(1)
                    get { window }.isEqualTo(Duration.ofSeconds(5))
                }
            get { all }
                .isNotNull()
                .and {
                    get { limit }.isEqualTo(2)
                    get { window }.isEqualTo(Duration.ofSeconds(10))
                }
        }
    }
}
